compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/CollectionStubMethodLowering.kt

```diff
@@ -172,13 +180,22 @@ private class CollectionStubMethodLowering(val context: JvmBackendContext) : Cla
 
     // Compute stubs that should be generated, compare based on signature
     private fun generateRelevantStubMethods(irClass: IrClass): Set<IrSimpleFunction> {
+
+        // For superTypes from java, we only generate stub if all further superTypes which are subtype of the immutable class are interfaces,
```

Comment:
Could you please move this logic into `comesFromJava`, rename that method accordingly and make it private, and also update the comment on lines 195-196?

Reply:
Done, didn't notice I forgot to change comment.